eta/core/models.py

```diff
@@ -255,16 +255,86 @@ def init_models_dir(new_models_dir):
     manifest.write_to_dir(new_models_dir)
 
 
-def recommend_paths_for_model(name):
-    '''Recommends a base filename and models directory for the given model
-    name, if possible.
+def publish_public_model(
+        name, google_drive_id, description=None, base_filename=None,
+        models_dir=None):
+    '''Publishes a new model to the public ETA model registry.
 
-    The recommendations are made by looking up the corresponding values from
-    the latest version of the model currently registered.
+    This function assumes that the model has been uploaded to Google Drive by
+    an ETA administrator and that you have the ID of the file to provide here.
+
+    This function performs the following actions:
+        - recommends a base filename and models directory, if necessary
+        - performs a dry run of the model registration process to check for
+            potential problems
+        - registers the model in the manifest of its models directory
+
+    The keyword arguments `base_filename` and `models_dir` configure the
+    filename (version information is added automatically) and models directory,
+    respectively, to use when downloading the model. If you are uploading a
+    new version of an existing model, these arguments can be omitted and the
+    same values are inherited from the most recent version of the model. If
+    this is a brand new model, these values are required
+
+    If the specified models directory does not have a models manifest file, one
+    is created. Note that new models directories are not automatically added to
+    your models search path, so models in a newly initialized directory will
+    not be findable until you add the directory to your models search path.
+
+    Args:
+        name: a name for the model, which can optionally have "@<ver>" appended
+            to assign a version to the model
+        google_drive_id: the ID of the model file in Google Drive
+        description: an optional description for the model
+        base_filename: an optional base filename to use when writing the model
+            to disk. By default, a value is inferred as explained above
+        models_dir: an optional directory in which to register the model. By
```

Comment:
This should include the note that the `models_dir` is required for a first time, new model upload.